pmlc/dialect/affinex/transforms/memref_dataflow_opt.cc

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 Intel Corporation
+
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "pmlc/dialect/affinex/transforms/pass_detail.h"
+
+using namespace mlir; // NOLINT
+
+struct MemAccess {
+  Value memref;
+  AffineMap map;
+  llvm::SmallVector<Value, 4> indices;
+  MemAccess() = default;
+};
+
+namespace llvm {
+template <>
+struct DenseMapInfo<MemAccess> {
+  static inline MemAccess getEmptyKey() {
+    MemAccess access;
+    access.memref = DenseMapInfo<mlir::Value>::getEmptyKey();
+    return access;
+  }
+  static inline MemAccess getTombstoneKey() {
+    MemAccess access;
+    access.memref = DenseMapInfo<mlir::Value>::getEmptyKey();
+    return access;
+  }
+  static unsigned getHashValue(const MemAccess &Val) {
```

Comment:
rename `Val` to `val` (the older style from LLVM is deprecated).

Reply:
Done